proxy/src/serverless/sql_over_http.rs

```diff
@@ -249,6 +251,44 @@ pub(crate) async fn handle(
     let mut response = match result {
         Ok(r) => {
             ctx.set_success();
+
+            // Handling the error response from local proxy here
+            if config.authentication_config.is_auth_broker && r.status().is_server_error() {
+                let status = r.status();
+
+                let body_bytes = r
+                    .collect()
+                    .await
+                    .map_err(|e| {
+                        ApiError::InternalServerError(anyhow::Error::msg(format!(
+                            "could not collect http body: {e}"
+                        )))
+                    })?
+                    .to_bytes();
+
+                if let Ok(json_map) =
+                    serde_json::from_slice::<IndexMap<&str, &RawValue>>(&body_bytes)
+                {
+                    let message = json_map.get("message");
+                    if let Some(message) = message {
+                        let msg: String = match serde_json::from_str(message.get()) {
+                            Ok(msg) => msg,
+                            Err(_) => {
+                                "Unable to parse the response message from server".to_string()
+                            }
+                        };
+
+                        error!("Error response from local_proxy: {status} {msg}");
+                        return json_response(status, json!({ "message": msg }));
+                    }
```

Comment:
I still think rather than just capturing a single message, we should instead remove undesired entries from `json_map`, and we can then return `json_response(state, json_map)`.

Undesired entries like our internal error message, or the error kind.

Need to figure out what key names we would use, but maybe we could use a catch all key prefix like `neon:` to indicate keys we want to keep (so we could use `json_map.retain(|key, _| !key.starts_with("neon:"))` for a simple impl until we start emitting and capturing these keys later)

Reply:
Thanks! Fixed